REBASING.md

```diff
@@ -16,18 +16,20 @@ Also, this guide assumes that the main remote repo of `zmap/dns` is called `orig
 3. Pull and merge in the latest changes from `miekg/dns`: `git pull miekg master`
     - At this point, there _shouldn't_ be any conflicts. Master is not the branch where the ZDNS patching is applied, so it should be a clean merge.
 4. Push the changes to `zmap/dns` remote: `git push origin  master`
-5. Now, switch to the `zdns` branch: `git checkout zdns`
-6. Ensure that you have pulled the latest changes from `zmap/dns`: `git pull origin zdns`
-7. Merge the new changes from `miekg/dns` into the `zdns` branch: `git merge master`
+5. Pull latest tags from `miekg/dns`: `git fetch miekg --all --tags`
+6. Run the following command to push the latest tag from `miekg/dns` to `zmap/dns`: `LATEST_TAG=$(git tag --merged | sort -V | tail -n 1); git push origin "${LATEST_TAG}"`
+7. Now, switch to the `zdns` branch: `git checkout zdns`
+8. Ensure that you have pulled the latest changes from `zmap/dns`: `git pull origin zdns`
+9. Merge the new changes from `miekg/dns` into the `zdns` branch: `git merge master`
     - This will likely result in conflicts.
     - The reason for choosing a merge vs. a rebase is twofold. 
         - First, it provides a more intuitive (though possibly more verbose) commit history, which may be useful as new devs or other inspect project history.
         - Second, and more importantly, it removes the need for the developer doing this process to fix conflicts at each of the 5 or 6 "patch" commits on the `zdns` branch. Because these changes occur in place that often modified by the `miekg` team, each commit ends up with a variety of merge conflicts and it becomes easy to incorrectly fix these conflicts. Merging instead ensures that only one set of conflicts need to be address, if any: the final version of the `miekg/dns` library with the patch that was made.
-8. Fix any conflicts that came about in the merge, then add and commit the output `git add --all; git commit -m "address conflicts between miekg and zmap dns libs"`
-9. Run tests to ensure that nothing has been broken: `go test github.com/zmap/dns`
-10. If all is successful, then push the changes to the `zmap/dns` branch: `git push origin zdns`
-12. If desired, remove the miekg remote: `git remote remove miekg`
-11. If needed, open a PR for the recent changes in the `zdns` branch.
+10. Fix any conflicts that came about in the merge, then add and commit the output `git add --all; git commit -m "address conflicts between miekg and zmap dns libs"`
+11. Run tests to ensure that nothing has been broken: `go test github.com/zmap/dns`
+12. If all is successful, then push the changes to the `zmap/dns` branch: `git push origin zdns`
+13. If desired, remove the miekg remote: `git remote remove miekg`
+14. If needed, open a PR for the recent changes in the `zdns` branch.
```

Comment:
If you're modifying `zdns` in the steps above, you can't really do this because the branch has already been mutated. We should edit the instructions to:
- Create a temporary branch based off of `zdns` prior to fetching the `miekg` changes
- merge `master` into the temporary branch
- Create PR to merge the temporary branch into ZDNS (effectively a fast-forward)
- Delete the temporary branch.

If we don't actually care to review these changes, then your instructions look good as they are.

Reply:
I was imagining a PR from a fork like `spencerdrak/dns` into `zmap/dns`, but likely there'll also be contributor working directly in `zmap/dns` at some point. I can modify the instructions to reflect this and also be more clear on what to do if you're working in a fork. Thanks!